src/EditorFeatures/Core/Implementation/EncapsulateField/AbstractEncapsulateFieldCommandHandler.cs

```diff
@@ -83,22 +82,30 @@ private bool Execute(EncapsulateFieldCommandArgs args, IUIThreadOperationScope w
 
                 var service = document.GetLanguageService<AbstractEncapsulateFieldService>();
 
-                var result = service.EncapsulateFieldAsync(document, spans.First().Span.ToTextSpan(), true, cancellationToken).WaitAndGetResult(cancellationToken);
+                // This is the last point where the operation can be canceled by the operation context
+                // managed by the command system. Make sure to not ignore a cancellation request which
+                // occurred prior to this line before proceeding with the Roslyn-managed dialogs.
+                cancellationToken.ThrowIfCancellationRequested();
 
                 // We are about to show a modal UI dialog so we should take over the command execution
                 // wait context. That means the command system won't attempt to show its own wait dialog 
                 // and also will take it into consideration when measuring command handling duration.
                 waitScope.Context.TakeOwnership();
```

Comment:
Woudln't it make more sense to leave it as it was, but only do the .TakeOwnership right before we show the preview dialog?

Reply:
Note: if we do that, we would still want to do this work before showing hte preview:

```
result.GetNameAsync(cancellationToken).WaitAndGetResult(cancellationToken),
result.GetGlyphAsync(cancellationToken).WaitAndGetResult(cancellationToken),
```

Reply:
agreed.  i would imagine we have this problem in a few other places.